reconcile/utils/rosa/rosa_cli.py

```diff
@@ -0,0 +1,193 @@
+            logger(f.read())
+
+    def cleanup(self) -> None:
+        os.remove(self.log_file)
+
+
+class RosaCliResult:
+    """
+    Represents the result of a ROSA CLI execution.
+    """
+
+    def __init__(
+        self,
+        status: JobStatus,
+        command: str,
+        log_handle: Optional[LogHandle] = None,
+    ) -> None:
+        self.status = status
+        self.command = command
+        self.log_handle = log_handle
+
+    def get_log_lines(self, max_lines: int = 5) -> list[str]:
+        if self.log_handle:
+            return self.log_handle.get_log_lines(max_lines)
+        return []
+
+    def write_logs_to_logger(self, logger: Callable[..., None]) -> None:
+        if self.log_handle:
+            self.log_handle.write_logs_to_logger(logger)
+
+
+class RosaCliException(Exception, RosaCliResult):
+    """
+    Represents an exception that occurred during a ROSA CLI execution.
+    """
+
+    def __init__(
+        self,
+        status: JobStatus,
+        command: str,
+        log_handle: Optional[LogHandle] = None,
+    ) -> None:
+        Exception.__init__(
+            self, f"ROSA CLI execution failed with status: {status}, cmd: {command}"
+        )
+        RosaCliResult.__init__(self, status, command, log_handle)
+
+
+class RosaJob(K8sJob, BaseModel, frozen=True):
+    """
+    Represents a ROSA CLI job. It leverages the reconcile.utils.jobcontroller module
+    functionality to execute ROSA CLI commands in a Kubernetes cluster.
+
+    Since the ROSA CLI requires access to both AWS and OCM, the job is executed
+    with the required credentials and tokens.
+
+    The credentials used for AWS are as temporary ones that expire after a
+    defined amount of time, usually 15 minutes. The OCM access token is provided
+    as a JWT with an expiration of 15 minutes.
+
+    Since both credential types are of temporary nature, they are passed to the
+    Kubernetes job as environment variables.
+    """
+
+    account_name: str
+    cluster_name: str
+    org_id: str
+    cmd: str
+
+    aws_credentials: AWSTemporaryCredentials
+    ocm_token: str
+
+    dry_run: bool = False
+
+    def name_prefix(self) -> str:
+        prefix = "rosa-cli"
+        if self.dry_run:
+            prefix += "-dry-run"
+        return prefix
+
+    def unit_of_work_identity(self) -> Any:
+        return {
+            "cmd": self.cmd,
+            "account_name": self.account_name,
+            "cluster_name": self.cluster_name,
+            "org_id": self.org_id,
+            "dry_run": self.dry_run,
+        }
+
+    def annotations(self) -> dict[str, str]:
+        return {
+            "qontract.rosa.account_name": self.account_name,
+            "qontract.rosa.cluster_name": self.cluster_name,
+            "qontract.rosa.org_id": self.org_id,
+        }
+
+    def job_spec(self) -> V1JobSpec:
+        return V1JobSpec(
+            backoff_limit=1,
+            ttl_seconds_after_finished=3600,
```

Comment:
we may want to keep the job longer so we have ample time to inspect logs in case of failures

Reply:
the job logs are transfered to stdout of the integration after either a timeout, a success or an error. but i wonder if the job and its pods would be interesting to keep for longer for any other debuging steps else but reading logs 🤔

Reply:
SGTM. Lets leave it as is and we can optimize later if needed